### ISSUE TITLE
Portion "units"

### ISSUE THREAD

**Issue by ModProg** [nextcloud/cookbook#2528]
**Is your feature request related to a problem? Please describe.**
When backing recipes often can be adjusted to the baking sheet you use, i.e. do you use a Springform 23cm vs 28cm. While it is obviously possible to calculate these by hand everytime you bake and enter the correct ratio in the portions field it isn't really convinient.

**Describe the solution you'd like**
Define a list of “backing units” with the correct ratio assigned.
When creating a backing recipe you can set the portions to "1 28cm Springform" and then be able to select e.g. a `23cm Springform` when I want to bake a smaller version.

**Describe alternatives you've considered**
A less convinient solution could also be to allow defining custom “steps” for each recipe's portion size selector, i.e. add a list of common “portion sizes” to a recipe, like "0.3" or "0.6".

**Comment by stefan123t** [comment]
I just used a basic calculation for the cylinder to calculate two Volumes with same height 5 cm for your given "Springform":

V = pi * ( D / 2 )² * h

* pi * (23 cm / 2)² * 5 cm = 2077,38 cm³ or ~ 2,0 l
* pi * (28 cm / 2)² * 5 cm = 3078,76 cm³ or ~ 3,0 l

It may get more complicate once you use a "Kranz" or even "Gugelhupf" form 🧁.
You will have to subtract something for the hole in the middle of such a "torus" form.

**Comment by ModProg** [comment]
> You will have to subtract something for the hole in the middle of such a "torus" form.

Or you just fill it with water and measure it.


But I don't mind doing these calculations my self, I just don't want to redo them every time, hence I'd like an option to create and manage such named portion sizes.

**Comment by stefan123t** [comment]
Yes measuring the volume in liters would be an easy way to get a good value.

So you would like to have a **volume** in liter / gallons for the dough and a name for that volume, e.g. `Springform ø23cm`.
These custom presets would probably be in addition to the common **yield** in kg or servings / loafs of bread.

I am still unsure how to apply / attach these custom serving presets to one of the recipes ?

These are actually general presets and not yet tied to a single recipe.
* Springform ø23 cm = 1.5 l (default)
* Springform ø26 cm = 1.75 l
* Springform ø28 cm = 2.0 l
* Gugelhupf ø20 cm = 1.6 l
* Gärkorb ø23 cm = 500 g
* Gärkorb ø27 cm = 1,0 kg
* Gärkorb länglich 44 cm = 500 g
* Gärkorb oval 30 cm = 500 g
* Gärkorb oval 36 cm = 750 g

Would you come up with the correct volumes / yields for a basic list of baking forms, so they can be added as defaults to the list ?
I just made up some numbers above as an example list, none of this is a real measurement.

But then using a Springform may make sense only in relation with a recipe, for which this relation has already been confirmed and linked or somehow validated as having the right yield. So the amounts given in the recipe have to be related to one of your baking forms / trays first.

This would require one to pre-select and set a default custom baking form / yield size to such a recipe.

**Comment by legheed** [comment]
I'd like to add that custom names for "portions" would be great. When you look at the example at https://schema.org/Recipe, it has "Yield: 1 loaf", so it should be possible to change this.

Maybe add a text field next to the portion amount field; if left empty, the result will just be "x portions", but if you add "loaf", it will be "x loaf".

This would also allow to hint at the intended portion sizes. E.g., when you have a recipe for Mousse au Chocolat for 50 portions, you should know how much a portion actually is. you could create the recipe, set portions to "50" and add "Nocks à 60 g"

**Comment by christianlupus** [comment]
Generally speaking, it is possible to store arbitrary text as part of the `recipeYield` field. So, yeah, it could be "1 leaf" without violating the standard. The problem comes with the parsing of such an information. We need the numeric value for the recalculation of receipe ingredients.

I personally do not (yet?) see an option to handle this UX-wise. Assume, we are baking. We have springforms 24 and 28cm. We have the volumes. We have a yield of "1" or "1 cake" (of 24cm) by default. Now, how would one switch to 28cm? Having a drop-down to select 28cm springform and roughly double all ingredient values? Wouldn't this be overwhelming if there are all kinds of tools/sizes/... Listed there?
When we go the route down, having something like "50 nocks, 60g each)" is even more problematic. This one is parametrical (the 60g might also be 50g or 65g). So, it could be mere informational (so the user knows is 60g each portion) or part of the calculation (then, we need a way to tackle this efficiently).

Note to self: Use non-text representation using QuantitiveValue to have KV pairs of additional values.

**Comment by legheed** [comment]
>>This one is parametrical (the 60g might also be 50g or 65g). So, it could be mere informational (so the user knows is 60g each portion) or part of the calculation (then, we need a way to tackle this efficiently).

I thought of it as mere informational, because to make sense of the number or portions needed I have to know how much a portion is supposed to be by the author of the recipe. At least for me this would be enough. Of course this is something that could easily go into the description of the recipe, but I thought it would be nice if portion size would be listed next to portion yield.
